packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -2366,7 +2366,27 @@ class RoundSliderThumbShape extends SliderComponentShape {
     final double evaluatedElevation = elevationTween.evaluate(activationAnimation);
     final Path path = Path()
       ..addArc(Rect.fromCenter(center: center, width: 2 * radius, height: 2 * radius), 0, math.pi * 2);
-    canvas.drawShadow(path, Colors.black, evaluatedElevation, true);
+
+    bool paintShadows = true;
+    assert(() {
```

Comment:
nit: move this to a helper method instead of repeating the same code below.

Reply:
Done-ish